src/wmtk/utils/OperationLogger.h

```diff
@@ -0,0 +1,102 @@
+#pragma once
+
+#include <oneapi/tbb/concurrent_vector.h>
+#include <oneapi/tbb/mutex.h>
+#if defined(__clang__)
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wcast-qual"
```

Comment:
You might want to use the `EnableWarnings.hpp` and `DisableWarnings.hpp` for this instead.